src/test/java/lambda/part1/exercise/Exercise1.java

```diff
@@ -0,0 +1,114 @@
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.arrayContaining;
+import static org.hamcrest.Matchers.is;
+
+class Exercise1 {
+
+    @Test
+    void sortPersonsByAgeUsingArraysSortLocalComparator() {
+        Person[] persons = getPersons();
+
+        Comparator<Person> personByAge = new Comparator<Person>() {
+            @Override
+            public int compare(Person pers1, Person pers2) {
+                return Integer.compare(pers1.getAge(), pers2.getAge());
+            }
+        };
+
+        Arrays.sort(persons, personByAge);
+
+        assertThat(persons, is(arrayContaining(
+                new Person("Иван", "Мельников", 20),
+                new Person("Николай", "Зимов", 30),
+                new Person("Алексей", "Доренко", 40),
+                new Person("Артем", "Зимов", 45)
+        )));
+    }
+
+    @Test
+    void sortPersonsByAgeUsingArraysSortAnonymousComparator() {
+        Person[] persons = getPersons();
+
+        Arrays.sort(persons, new Comparator<Person>() {
+            @Override
+            public int compare(Person pers1, Person pers2) {
+                return Integer.compare(pers1.getAge(), pers2.getAge());
+            }
+        });
+
+        assertThat(persons, is(arrayContaining(
+                new Person("Иван", "Мельников", 20),
+                new Person("Николай", "Зимов", 30),
+                new Person("Алексей", "Доренко", 40),
+                new Person("Артем", "Зимов", 45)
+        )));
+    }
+
+    @Test
+    void sortPersonsByLastNameThenFirstNameUsingArraysSortAnonymousComparator() {
+        Person[] persons = getPersons();
+
+        Arrays.sort(persons, new Comparator<Person>() {
+            @Override
+            public int compare(Person pers1, Person pers2) {
+                int sortByLastName =  pers1.getLastName().compareTo(pers2.getLastName());
+                int sortByFirstName = pers1.getFirstName().compareTo(pers2.getFirstName());
+                return sortByLastName == 0 ? sortByFirstName : sortByLastName;
+            }
+        });
+
+
+        assertThat(persons, is(arrayContaining(
+                new Person("Алексей", "Доренко", 40),
+                new Person("Артем", "Зимов", 45),
+                new Person("Николай", "Зимов", 30),
+                new Person("Иван", "Мельников", 20)
+        )));
+    }
+
+    @Test
+    void findFirstWithAge30UsingGuavaPredicate() {
+        List<Person> persons = Arrays.asList(getPersons());
+
+        Predicate<Person> agePredicate = person -> person.getAge() == 30;
+
+        Person person = Iterables.find(persons, agePredicate);
+
+        assertThat(person, is(new Person("Николай", "Зимов", 30)));
+    }
+
+    @Test
+    void findFirstWithAge30UsingGuavaAnonymousPredicate() {
+        List<Person> persons = Arrays.asList(getPersons());
+
+        // TODO use FluentIterable
+        Person person = Iterables.find(persons, new Predicate<Person>() {
```

Comment:
Why you use Iterables? I think your version is correct, but you need to use FluentIterable?